### DISCUSSION TITLE
Requesting special app permissions from mobile users

### DISCUSSION THREAD

**Discussion by Longersking** [termux/termux-app#3725] [Ideas]
I discovered that on the phone, there's a special app permissions menu in the app management settings. Here, it displays the following permissions: Modify system settings, device and app notifications, usage access, access to all files, WLAN control, and floating window. Among these special app permissions, I found that only Termux requested floating window and notification permissions. I'm wondering if Termux attempts to request these permissions, could it potentially have more functionalities under non-root permissions (such as proot)? This would only require requesting these special permissions at the time of activation, similar to how system-native file managers need to request access to all files when accessing certain directories.

**Comment by sylirre** [comment]
Proot and command line utilities in general can only inherit permissions that Android implements through Linux groups. Particularly these are Internet and storage access permissions. Rest is doable only through Android app bridge such as Termux:API. The latter can be used by `proot` when necessary directory binding and environment variables are set up (already implemented in `proot-distro` package).
